tests/functional/mesh/mesh_test.go

```diff
@@ -94,6 +99,56 @@ func TestMeshConnections(t *testing.T) {
 			if connectionsReady == false {
 				t.Error("Timed out while waiting for connections")
 			}
+			mesh.Shutdown()
+		})
+	}
+}
+
+// Test that a mesh starts and that connections are what we expect
+func TestMeshShutdown(t *testing.T) {
+	testTable := []struct {
+		filename string
+	}{
+		{"mesh-definitions/random-mesh-tcp.yaml"},
+		{"mesh-definitions/random-mesh-udp.yaml"},
+		{"mesh-definitions/random-mesh-ws.yaml"},
+	}
+	for _, data := range testTable {
+		filename := data.filename
+		t.Run(filename, func(t *testing.T) {
+			mesh, err := NewMeshFromFile(filename)
+			if err != nil {
+				t.Error(err)
+			}
+			err = mesh.WaitForReady(10000)
+			if err != nil {
+				t.Error(err)
+			}
+			mesh.Shutdown()
+			mesh.WaitForShutdown()
+
+			// Check that the connections are closed
+			pid := os.Getpid()
+			pidString := strconv.Itoa(pid)
+			done := false
+			for timeout := 1000000; timeout > 0 && !done; {
+				var out bytes.Buffer
+				cmd := exec.Command("netstat", "-tunp")
+				cmd.Stdout = &out
+				err := cmd.Run()
+				if err != nil {
+					t.Fatal(err)
+				}
+				if !strings.Contains(out.String(), pidString) {
+					done = true
+					break
+				}
+				time.Sleep(100 * time.Millisecond)
+				timeout -= 100
```

Comment:
You're sleeping for 100 milliseconds but then only subtracting 100 nanoseconds.  Did you intend to do `timeout -= 100 * time.Millisecond`?